### PR TITLE
Geometry Transforms: avoid division by zero & some degenerate cases

### PR DESCRIPTION
**Pull request by cv3d**
## Description
Minor update to avoid returning nans due to division by zeros & some degenerate cases

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [x] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by stefanv** [comment]
Hey, @cv3d, thank you for submitting this PR.

Could you please expand on the problem you ran into, and why this is the correct fix?

**Comment by cv3d** [comment]
Hi @stefanv. I am using these geometry transforms in a RANSAC iteration for model estimation, where some degenerative cases might be the cause of such errors.

The basic idea of this PR is to return some numerical values (even if they are very large) rather than NaNs, which might break the user code.

**Comment by cv3d** [comment]
@stefanv The failing test is NOT related to this PR. Shall I do anything?

**Comment by jni** [comment]
@cv3d I think we would like some comments around those lines to explain their purpose, as well as a test case that would fail without the fixes (so that if anyone changes the logic or the algorithm in the future, the new algorithm still has the same stability).

**Comment by cv3d** [comment]
@jni How about now?

File: skimage/transform/_geometric.py [hunk 1]
```diff
@@ -549,9 +549,10 @@ def _apply_mat(self, coords, matrix):
         src = np.vstack((x, y, np.ones_like(x)))
         dst = src.T @ matrix.T
 
+        # avoid division by zero
+        dst[dst[:, 2] == 0, 2] = np.finfo(float).eps
         # rescale to homogeneous coordinates
-        dst[:, 0] /= dst[:, 2]
-        dst[:, 1] /= dst[:, 2]
+        dst[:, :2] /= dst[:, 2:3]
```
Comment (emmanuelle) [inline skimage/transform/_geometric.py]: why not ``dst[:, :2] /= dst[:, 2]``? Broadcasting should work here I think?

Comment (cv3d) [reply skimage/transform/_geometric.py]: @emmanuelle Not really. `dst[:, :2]` dimension is `(n, 2)`, `dst[:, 2:3]` has a dimension of `(n,1)`, while `dst[:, 2]` dimension is just `n`.

**Comment by cv3d** [comment]
@jni All done!

File: skimage/transform/_geometric.py [hunk 2]
```diff
@@ -549,9 +549,12 @@ def _apply_mat(self, coords, matrix):
         src = np.vstack((x, y, np.ones_like(x)))
         dst = src.T @ matrix.T
 
+        # below, we will divide by the last dimension of the homogeneous
+        # coordinate matrix. In order to avoid division by zero,
+        # we replace exact zeros in this column with a very small number.
```
Comment (jni) [inline skimage/transform/_geometric.py]: :clap: good stuff!

**Comment by jni** [comment]
@cv3d thanks for the update! Looks great!

@scikit-image/core imho this is merge-ready!

**Comment by stefanv** [comment]
Thanks, @cv3d!

**Pull request merged**